optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/ConstraintCollectorsTest.java

```diff
@@ -0,0 +1,127 @@
+package org.optaplanner.core.api.score.stream;
+
+import java.util.Comparator;
+
+import org.junit.Test;
+import org.optaplanner.core.api.score.stream.uni.UniConstraintCollector;
+
+import static org.junit.Assert.assertEquals;
+
+public class ConstraintCollectorsTest {
+
+    private <A, B, C> Runnable accumulate(UniConstraintCollector<A, B, C> collector, Object container, A value) {
+        return collector.accumulator().apply((B) container, value);
+    }
+
+    private <A, B, C> void assertResult(UniConstraintCollector<A, B, C> collector, Object container, C expectedResult) {
+        C actualResult = collector.finisher().apply((B) container);
+        assertEquals("Collector (" + collector + ") did not produced expected result.", expectedResult, actualResult);
+    }
+
+    @Test
+    public void maxComparable() {
```

Comment:
in ConstraintCollectors the order is "min, max", here the order is "max, min".
Can we keep the same order?

Can we also add the
     // ************************************************************************
    // max
    // ************************************************************************

Can we open an issue to also test the others (count, countDistinct, sum, etc)?